src/engraving/automation/automationtypes.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <map>
+
+namespace mu::engraving {
+struct AutomationPoint {
+    enum class InterpolationType {
+        Linear = 0,
+        Exponential,
+    };
+
+    double value = 0.;
+    InterpolationType interpolation = InterpolationType::Linear;
+
+    bool isValid() const
+    {
+        return id.isValid();
+    }
+};
+
+struct AutomationCurve {
+    enum class Type {
+        Unknown = 0,
+        Dynamics,
+    };
+
+    Type type = Type::Unknown;
```

Comment:
Eventually I think we'll want to have a map from type to curve, but for now this is probably fine